pkg/services/datasources/service/datasource.go

```diff
@@ -534,21 +534,26 @@ func (s *Service) UpdateDataSource(ctx context.Context, cmd *datasources.UpdateD
 			}
 		}
 
-		// TODO: we will eventually remove this check for moving the resource to it's separate API
-		if s.features != nil && s.features.IsEnabled(ctx, featuremgmt.FlagTeamHttpHeaders) && !cmd.OnlyUpdateLBACRulesFromAPI {
+		// TODO: Refactor to store lbac rules separate from a datasource
+		updatingLBACRules := cmd.OnlyUpdateLBACRulesFromAPI && s.features != nil && s.features.IsEnabled(ctx, featuremgmt.FlagTeamHttpHeaders) && cmd.JsonData != nil && cmd.JsonData.Get("teamHttpHeaders").Interface() != nil
+		if updatingLBACRules {
```

Comment:
The log message should be in the else statement.

Reply:
true dat!

Reply:
my bad! 👍